app/src/main/java/com/example/plantsapp/presentation/ui/utils/ContextExt.kt

```diff
@@ -20,13 +22,36 @@ fun Context.getCameraImageOutputUri(): Uri? {
         null
     }
 
-    return photoFile?.let {
-        FileProvider.getUriForFile(
-            this,
-            BuildConfig.APPLICATION_ID,
-            it
-        )
+    return photoFile?.let { getFileUri(it) }
+}
+
+fun Context.saveImageInAppStorage(imageUri: Uri): Uri? {
+    val newFile = try {
+        val file = createImageFile(this)
+
+        val inputStream = contentResolver.openInputStream(imageUri)
+        val outputStream = FileOutputStream(file)
+
+        copyStream(inputStream!!, outputStream)
+
+        outputStream.close()
+        inputStream.close()
+
+        file
+    } catch (exception: IOException) {
+        Timber.e(exception)
+        null
```

Comment:
How about throw here exception, and make this function non-nullable?

Reply:
Also, leave here 
//TODO Move this code to SaveUriToStorageUseCase

Reply:
https://github.com/Lonexera/students-lab/projects/1#card-73123534

Also I've created a task for this